pkg/reconciler/tenancy/workspacemounts/workspacemounts_reconcile_updater.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2024 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workspacemounts
+
+import (
+	"context"
+
+	"github.com/kcp-dev/logicalcluster/v3"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/sdk/apis/tenancy/v1alpha1"
+)
+
+const (
+	// WorkspaceShardHashAnnotationKey keeps track on which shard LogicalCluster must be scheduled. The value
+	// is a base36(sha224) hash of the Shard name.
+	WorkspaceShardHashAnnotationKey = "internal.tenancy.kcp.io/shard"
+)
+
+// workspaceStatusUpdater updates the status of the workspace based on the mount status.
+// TODO(mjudeikis): For now this only updates the status of the mount annotation. In the future
+// we should add second reconciler that will update the status of the workspace so triggering
+// it to be "not ready" if the mount is not ready.
+type workspaceStatusUpdater struct {
+	getMountObject func(ctx context.Context, cluster logicalcluster.Path, ref *v1.ObjectReference) (*unstructured.Unstructured, error)
+}
+
+func (r *workspaceStatusUpdater) reconcile(ctx context.Context, workspace *tenancyv1alpha1.Workspace) (reconcileStatus, error) {
+	if workspace.Annotations == nil {
+		workspace.Annotations = map[string]string{}
```

Comment:
doesn't this lead to hotlooping? Would move it down where you actually write to annotations.

Reply:
It should not. But optimized either way :)